### ISSUE TITLE
Feature Request: Ability to supply additional Header

### ISSUE THREAD

**Issue by peanutbother** [JohnPeel/api-client#1]
First of all: great library!
I'm using it in one of my libraries right now and I have come across a limitation:

For some routes I need to supply additional Header values. For now I implement it by matching url in the request method which isn't ideal.

My suggestion would be to introduce a `Header<T>` parameter in the call signature of the macro.
I'm willing to provide help if wanted.

kindly regards

**Comment by JohnPeel** [comment]
I'll look into it. Thanks for the feature request.

**Comment by JohnPeel** [comment]
We do run into a few issues, at least with how it's currently designed.

 * You can't really represent `HeaderName` as an identifier.
   * This makes it hard to support as a parameter name, or as an extension of the function body.
   * Possible workaround: We could use some modifications to convert identifiers to `HeaderName`, such as `x_request_id => X-REQUEST-ID`.
 * Converting T to `HeaderValue` is a bit annoying, but possible. It might introduce an Error type to accomidate both reqwest errors and http errors.
 
 Another possible workaround is to rewrite using a procedural macro. Though that would take some time.

**Comment by JohnPeel** [comment]
I've added the ability to do headers in the method body, but no custom headers yet.

If the header is defined in https://docs.rs/reqwest/latest/reqwest/header/index.html#constants, you can use the constant to assign the header a value.

Here is an example
```rust
    mod headers {
        use crate::{api, Api};

        api!(pub struct HeaderTest);

        const BASE_URL: &str = "https://ifconfig.me";

        impl HeaderTest {
            pub fn new() -> Self {
                Api::new()
            }

            api! {
                pub fn get_ua(ua: &str) -> String {
                    GET "{BASE_URL}/ua"
                    USER_AGENT: "{ua}"
                }
            }
        }
    }

    #[tokio::test]
    async fn example_header() {
        let api = HeaderTest::new();

        assert_eq!(
            api.get_ua("Api-client 0.1").await.unwrap(),
            "Api-client 0.1"
        );
    }
```

You can still use `pre_request` to add headers to any request as well.

**Comment by JohnPeel** [comment]
I'm planning to add custom headers as well, it just requires some more thought and testing.

**Comment by JohnPeel** [comment]
Okay, I've released v0.2.1, is now supports string literals to specify the header name, and any constant that implements `IntoHeaderName` such as the constants from `http::header`. The constant needs to be in-scope now though.
